ras_party/controllers/account_controller.py

```diff
@@ -669,3 +669,15 @@ def get_cases_for_casegroups(casegroup_ids, case_party_id):
     logger.debug('Successfully retrieved cases for casegroups',
                  case_party_id=case_party_id)
     return matching_cases
+
+
+@with_db_session
+def resend_verification_email_expired_token(token, session):
+    email_address = decode_email_token(token, duration=None)
+    respondent = query_respondent_by_email(email_address, session)
+
+    if not respondent:
+        raise ClientError("Respondent does not exist", status=404)
+
+    response = resend_verification_email(respondent.party_uuid)
```

Comment:
Not sure why we are going via resend_verification_email , this does a query for respondent by UUID which I THINK is redundant if we have already got got the respondent by email ?  If my assumption is correct then could we split out the last if else into its own private function and call it from both resend_verification_email_expired_token and resend_verification_email ( also possibly change the name of resend_verification_email to resend_verification_email_by_uuid ?)

Reply:
Updated to remove the second unnecessary call to the db.